src/Symfony/Component/ExpressionLanguage/Resources/bin/generate_operator_regex.php

```diff
@@ -15,9 +15,13 @@
 
 $regex = [];
 foreach ($operators as $operator => $length) {
-    // an operator that ends with a character must be followed by
-    // a whitespace or a parenthesis
-    $regex[] = preg_quote($operator, '/').(ctype_alpha($operator[$length - 1]) ? '(?=[\s(])' : '');
+    // Collisions of character operators:
+    // - an operator that begins with a character must not be following after dot
+    // - an operator that ends with a character must be followed by a whitespace or a parenthesis
+    $regex[] =
+        (ctype_alpha($operator[$length - 1]) ? '(?<!\.)' : '')
```

Comment:
should be `$operator[0]` as we want to test the first character as per the comment.

Reply:
the test should be that a whitespace must be before the operator to cover other cases like `foo|not in ...`. So the regex should be `(?=[\s()])`.

Reply:
>the test should be that a whitespace must be before the operator to cover other cases

Using whitespace incorrect work on case when operator in the start of expression - `not foo` .
Test https://github.com/symfony/symfony/pull/35707/files#diff-20d6647365cfcd9f6114d806f45a4771R184
I don't known Twig lexer, I will check this case.

Reply:
Wrong copy/pasting on the regex, should be `(?<=[\s(])`

Reply:
`(?<=[\s(])not(?=[\s(])` not matched first not in case `not foo.not or not bar.not`

Reply:
`(?<=^|[\s(])` - works. I will fix it. Thank you.